src/metagen/src/shared/files.rs

```diff
@@ -0,0 +1,62 @@
+use std::{borrow::Cow, collections::HashMap};
+
+use crate::interlude::*;
+use common::typegraph::{
+    visitor::{Edge, PathSegment},
+    visitor2::{self, NearestFn, VisitNext},
+    Typegraph,
+};
+
+#[derive(Debug)]
+pub struct ValuePath(pub Vec<Cow<'static, str>>);
+
+fn serialize_path_segment(seg: &PathSegment) -> Result<Cow<'static, str>> {
```

Comment:
Why did you decide against just outputting the ValuePathSegment literals instead of the parsed approach?

Reply:
What I meant to say is that you get to avoid the parsing code if you just output literals directly. The `ValuePath` enum in the rust case.